### ISSUE TITLE
IgnoreNullValues does not work with non-nullable value types

### ISSUE THREAD

**Issue by youcefnb** [dotnet/runtime#30795]
Setting JsonSerializerOptions property of IgnoreNullValues to true does not work. 

When attempting to deserialize a DateTime value that is null in the json string it throws the exception below. I thought setting the option to IgnoreNullValues should ignore the property, no?

converted to System.DateTime. Path: $.value[0].deletedDateTime 
InvalidOperationException: Cannot get the value of a token type 'Null' as a string.

Thanks
Youcef

**Comment by layomia** [comment]
Punting this to future, as we don't want to loosen this behavior before designing upcoming validation features including [required properties](https://github.com/dotnet/runtime/issues/29861) and [(de)serialization callbacks](https://github.com/dotnet/runtime/issues/1562).

**Comment by chris-rogala** [comment]
Today I ran into an issue with this with the a `DateTime` property. Basically, my object is something like:

```
public class SomeDTO
{
    public DateTime SomeAwesomeDate { get; set; }
}
```

When the body looks like this:

```
{
  "someAwesomeDate": null
}
```

And the method looks like this:

`public Task<IActionResult> Post([FromBody] SomeDTO model)`

The `model` is always `null`. I try the same thing with a `double`, and it reacts the same way with a `double`. Keep in mind, this is for a more complex contract and even if all the other properties are set, it will set the `model` from the body to null. This is using .NET Core 3.1. Thoughts?

**Comment by layomia** [comment]
> JsonIgnoreCondition doesn't seem to specificy a case for ignoring null json tokens on read.

In API review we determined that there was no good scenario for ignoring `null` tokens on deserialization (superseding https://github.com/dotnet/runtime/issues/30795#issuecomment-530909280) Performance was considered i.e. saving some calls to setters, but deemed insufficient as justification for the feature. So we corrected the mistake of 3.0 semantics by not including the feature in `JsonIgnoreCondition`.

**Comment by eiriktsarpalis** [comment]
Rather than extending functionality of an obsoleted property, we should look to providing more extensibility points so that users can have a greater deal of control over what properties should be ignored and when. This was expressed initially in #55781 but ultimately we'd be looking at addressing this via #63686.
